Tutorials update (#136)
coreos/fedora-coreos-docs: modules/ROOT/pages/tutorial-basic-provisioning.adoc

      - name: autologin-core.conf
        contents: |
          [Service]
          # Override Execstart in main unit
          ExecStart=
          # Add new Execstart with `-` prefix to ignore failure`
          ExecStart=-/usr/sbin/agetty --autologin core --noclear %I $TERM
storage:
  files:
    - path: /etc/hostname
      mode: 0644
      contents:
        inline: |
          tutorial
    - path: /etc/sysctl.d/20-silence-audit.conf
      mode: 0644
      contents:
        inline: |
          # Raise console message logging level from DEBUG (7) to WARNING (4)
          # to hide audit messages from the interactive console
          kernel.printk=4
----

This configuration can then be converted into an Ignition config with `fcct`:

[source,bash]
----
$ fcct --pretty --strict fcct-simple.yaml --output simple.ign
----

The resulting Igntion configuration produced by `fcct` as `simple.ign` has the following content:

[source,json]
----
{
  "ignition": {
    "version": "3.0.0"
  },
  "passwd": {
    "users": [
      {
        "name": "core",
        "passwordHash": "$6$xyz$QdWnVhA9Jebjwi2.aD6Ly.KW69PwO5EGuibLjRJJD0W6WnB3HQvkGo583.VFK9X0vAjnP9YImpHqvDtuj2M.s/"
      }
    ]

>   },

Reviewer: This is what the generated `ign` file looks like of the FCCT yaml above:

---
{
  "ignition": {
    "version": "3.0.1"
  },
  "storage": {
    "files": [
      {
        "path": "/etc/hostname",
        "contents": {
          "source": "data:,tutorial%0A"
        },
        "mode": 420
      },
      {
        "path": "/etc/sysctl.d/20-silence-audit.conf",
        "contents": {
          "source": "data:,%23%20Raise%20console%20message%20logging%20level%20from%20DEBUG%20(7)%20to%20WARNING%20(4)%0A%23%20to%20hide%20audit%20messages%20from%20the%20interactive%20console%0Akernel.printk%3D4%0A"
        },
        "mode": 420
      }
    ]
  },
  "systemd": {
    "units": [
      {
        "dropins": [
          {
            "contents": "[Service]\n# Override Execstart in main unit\nExecStart=\n# Add new Execstart with `-` prefix to ignore failure`\nExecStart=-/usr/sbin/agetty --autologin core --noclear %I $TERM\n",
            "name": "autologin-core.conf"
          }
        ],
        "name": "serial-getty@ttyS0.service"
      }
    ]
  }
}
---

>   },

Author: My bad I forgot to update this one after a test :facepalm:

Fedora CoreOS 32.20200715.3.0
Kernel 5.7.8-200.fc32.x86_64 on an x86_64 (ttyS0)

SSH host key: SHA256:XlbayjbgDKNoAAHQxsEL5Q7BdwLxxWSw4NXN9SALLmo (ED25519)
SSH host key: SHA256:3sx5jseteO4BvdOMWIi0J4koQL015mLonnD0UPTtnZk (ECDSA)
SSH host key: SHA256:K0fn5/TMJOoMs7Fu7RRkE7IBEf2t8OYCfVaVc+GJWGs (RSA)
ens2: 192.168.122.127 fe80::5054:ff:feb9:3d97
Ignition: user provided config was applied
No ssh authorized keys provided by Ignition or Afterburn
tutorial login: core (automatic login)

[core@tutorial ~]$
----

Let's verify that our configuration has been correctly applied. As we were automatically logged in to the terminal, we can safely assume that the systemd dropin has been created:

[source,bash]
----
[core@tutorial ~]$ systemctl cat serial-getty@ttyS0.service
# /usr/lib/systemd/system/serial-getty@.service
...

# /etc/systemd/system/serial-getty@ttyS0.service.d/autologin-core.conf
[Service]
# Override Execstart in main unit
ExecStart=
# Add new Execstart with `-` prefix to ignore failure`
ExecStart=-/usr/sbin/agetty --autologin core --noclear %I $TERM
----

We can also check that the hostname has correctly been set:

[source,bash]
----
[core@tutorial ~]$ hostname
tutorial

> ----

Reviewer 2: Might be cool if we go a little more explicit here. So we first check the file got written

```
$ cat /etc/hostname
tutorial
```

Then run `hostnamectl` which is a little more fancy:

```
$ hostnamectl 
   Static hostname: n/a
Transient hostname: foo.example.com
         Icon name: computer-desktop
           Chassis: desktop
        Machine ID: acc83b1c3cd04189897696fe50fd1932
           Boot ID: 5e14930ff3504e3da0fb26f6325c0cb6
  Operating System: Fedora CoreOS 32.20200726.2.0
       CPE OS Name: cpe:/o:fedoraproject:fedora:32
            Kernel: Linux 5.7.10-201.fc32.x86_64
      Architecture: x86-64
```

       Docs: https://github.com/coreos/zincati
   Main PID: 889 (zincati)
      Tasks: 2 (limit: 2288)
     Memory: 14.2M
     CGroup: /system.slice/zincati.service
             └─889 /usr/libexec/zincati agent -v

Aug 06 14:50:36 tutorial systemd[1]: Started Zincati Update Agent.
Aug 06 14:50:36 tutorial zincati[889]: [INFO ] starting update agent (zincati 0.0.12)
Aug 06 14:50:39 tutorial zincati[889]: [INFO ] Cincinnati service: https://updates.coreos.fedoraproject.org
Aug 06 14:50:39 tutorial zincati[889]: [INFO ] agent running on node 'dbe8968f75c34d9eb3d3c4c226aa8fdf', in update group 'default'
Aug 06 14:50:39 tutorial zincati[889]: [INFO ] initialization complete, auto-updates logic enabled
----

One other interesting thing to do is view the logs from Ignition in case there is anything interesting there you may want to investigate:

[source,bash]
----
[core@tutorial ~]$ journalctl -t ignition | cat
...
----

And finally, of course you can use the podman (or Docker) command to inspect the current state of containers on the system:

[source,bash]
----
[core@tutorial ~]$ podman version
[core@tutorial ~]$ podman info
----

> *Note:* `podman` commands can be run as root or as non-root user. However you will need use `sudo` for `docker` commands as the core user is not part of the `docker` group by default.

Reviewer 2: ```suggestion
NOTE: `podman` commands can be run as root or as non-root user. `docker` commands need to be run as root via `sudo` unless the user has been added to the `docker` group.
```

Asciidoc picks up this and makes a nicely formatted box for it if it starts with `NOTE:`

> *Note:* Running `docker` daemon and `podman` at the same time can cause issues and result in expected behaviour. Here's an https://docs.fedoraproject.org/en-US/fedora-coreos/faq/#_can_i_run_containers_via_docker_and_podman_at_the_same_time[FAQ Entry] describing the issue.

Reviewer 2: ```suggestion
NOTE: Running containers via `docker` and `podman` at the same time can cause issues and result in unexpected behaviour. Refer to the https://docs.fedoraproject.org/en-US/fedora-coreos/faq/#_can_i_run_containers_via_docker_and_podman_at_the_same_time[FAQ Entry] for more details.
```
